docs/docs/guides/summarization.md

```diff
@@ -19,43 +19,31 @@ from langmem.short_term import summarize_messages, RunningSummary
 from langchain_openai import ChatOpenAI
 
 model = ChatOpenAI(model="gpt-4o")
-# NOTE: we're also setting max output tokens for the summary
-# this should match max_summary_tokens in `summarize_messages` for better
-# token budget estimates
 # highlight-next-line
-summarization_model = model.bind(max_tokens=128)
+summarization_model = model.bind(max_tokens=128)  # (1)!
 
 # We will keep track of our running summary in the graph state
 class SummaryState(MessagesState):
     summary: RunningSummary | None
 
 # Define the node that will be calling the LLM
 def call_model(state: SummaryState) -> SummaryState:
-    # We will attempt to summarize messages before the LLM is called
-    # If the messages in state["messages"] fit into max tokens budget,
-    # we will simply return those messages. Otherwise, we will summarize
-    # and return [summary_message] + remaining_messages
     # highlight-next-line
-    summarization_result = summarize_messages(
+    summarization_result = summarize_messages(  # (2)!
         state["messages"],
-        # IMPORTANT: Pass running summary, if any. This is what
-        # allows summarize_messages to avoid re-summarizing the same
-        # messages on every conversation turn
+        # IMPORTANT: Pass running summary, if any
         # highlight-next-line
-        running_summary=state.get("summary"),
-        # by default this is using approximate token counting,
-        # but you can also use LLM-specific one, like below
+        running_summary=state.get("summary"),  # (3)!
         # highlight-next-line
         token_counter=model.get_num_tokens_from_messages,
         model=summarization_model, 
-        max_tokens=256,
+        max_tokens=256,  # (4)!
```

Comment:
So just skiming, the logic is:
1. accumulate tokens until `max_tokens_before_summary`
2. truncate the history back down to `max_tokens`
3 generate a summary of <= `max_summary_tokens`

So the resulting token count would be ~ max_tokens + max_summary_tokens ?

Reply:
no, it is

1. accumulate tokens until >= max_tokens_before_summary
2. generate a summary of up to max_tokens_before_summary
3. combine with remaining messages

the resulting message history should be within `max_tokens`